x-pack/plugins/alerting/server/rule_type_registry_deprecated_consumers.ts

```diff
@@ -0,0 +1,75 @@
+ */
+
+import { ALERTS_FEATURE_ID } from './types';
+
+const ruleTypeIdWithLegacyConsumers: Record<string, string[]> = {
+  'example.always-firing': [ALERTS_FEATURE_ID],
+  transform_health: [ALERTS_FEATURE_ID],
+  '.index-threshold': [ALERTS_FEATURE_ID],
+  '.geo-containment': [ALERTS_FEATURE_ID],
+  '.es-query': [ALERTS_FEATURE_ID, 'discover'],
+  'xpack.ml.anomaly_detection_alert': [ALERTS_FEATURE_ID],
+  'xpack.ml.anomaly_detection_jobs_health': [ALERTS_FEATURE_ID],
+  'xpack.synthetics.alerts.monitorStatus': [ALERTS_FEATURE_ID],
+  'xpack.synthetics.alerts.tls': [ALERTS_FEATURE_ID],
+  'xpack.uptime.alerts.monitorStatus': [ALERTS_FEATURE_ID],
+  'xpack.uptime.alerts.tlsCertificate': [ALERTS_FEATURE_ID],
+  'xpack.uptime.alerts.durationAnomaly': [ALERTS_FEATURE_ID],
+  'xpack.uptime.alerts.tls': [ALERTS_FEATURE_ID],
+  'siem.eqlRule': [ALERTS_FEATURE_ID],
+  'siem.savedQueryRule': [ALERTS_FEATURE_ID],
+  'siem.indicatorRule': [ALERTS_FEATURE_ID],
+  'siem.mlRule': [ALERTS_FEATURE_ID],
+  'siem.queryRule': [ALERTS_FEATURE_ID],
+  'siem.thresholdRule': [ALERTS_FEATURE_ID],
+  'siem.newTermsRule': [ALERTS_FEATURE_ID],
+  'siem.notifications': [ALERTS_FEATURE_ID],
+  'slo.rules.burnRate': [ALERTS_FEATURE_ID],
+  'metrics.alert.anomaly': [ALERTS_FEATURE_ID],
+  'logs.alert.document.count': [ALERTS_FEATURE_ID],
+  'metrics.alert.inventory.threshold': [ALERTS_FEATURE_ID],
+  'metrics.alert.threshold': [ALERTS_FEATURE_ID],
+  monitoring_alert_cluster_health: [ALERTS_FEATURE_ID],
+  monitoring_alert_license_expiration: [ALERTS_FEATURE_ID],
+  monitoring_alert_cpu_usage: [ALERTS_FEATURE_ID],
+  monitoring_alert_missing_monitoring_data: [ALERTS_FEATURE_ID],
+  monitoring_alert_disk_usage: [ALERTS_FEATURE_ID],
+  monitoring_alert_thread_pool_search_rejections: [ALERTS_FEATURE_ID],
+  monitoring_alert_thread_pool_write_rejections: [ALERTS_FEATURE_ID],
+  monitoring_alert_jvm_memory_usage: [ALERTS_FEATURE_ID],
+  monitoring_alert_nodes_changed: [ALERTS_FEATURE_ID],
+  monitoring_alert_logstash_version_mismatch: [ALERTS_FEATURE_ID],
+  monitoring_alert_kibana_version_mismatch: [ALERTS_FEATURE_ID],
+  monitoring_alert_elasticsearch_version_mismatch: [ALERTS_FEATURE_ID],
+  monitoring_ccr_read_exceptions: [ALERTS_FEATURE_ID],
+  monitoring_shard_size: [ALERTS_FEATURE_ID],
+  'apm.transaction_duration': [ALERTS_FEATURE_ID],
+  'apm.anomaly': [ALERTS_FEATURE_ID],
+  'apm.error_rate': [ALERTS_FEATURE_ID],
+  'apm.transaction_error_rate': [ALERTS_FEATURE_ID],
+};
+
+const isRuleTypeIdHasLegacyConsumers = (ruleTypeId: string): boolean => {
+  if (ruleTypeIdWithLegacyConsumers[ruleTypeId]) {
+    return true;
+  } else if (ruleTypeId.startsWith('test.')) {
+    return true;
+  }
+  return false;
+};
+
+const getRuleTypeIdLegacyConsumers = (ruleTypeId: string): string[] => {
+  if (ruleTypeIdWithLegacyConsumers[ruleTypeId]) {
+    return ruleTypeIdWithLegacyConsumers[ruleTypeId];
+  } else if (ruleTypeId.startsWith('test.')) {
+    return [ALERTS_FEATURE_ID];
+  }
```

Comment:
**nit:** can we add the rest of the rule types instead of the `test.` exception? It could cause more oddities down the road if ever someone prefixes their rule type with this and we want to move away from `alerts` consumer.

The full list of `test.*` rule types is:

```
  'test.always-firing',
  'test.always-firing-alert-as-data',
  'test.authorization',
  'test.cancellableRule',
  'test.cumulative-firing',
  'test.exceedsAlertLimit',
  'test.failing',
  'test.gold.noop',
  'test.longRunning',
  'test.multipleSearches',
  'test.never-firing',
  'test.noop',
  'test.onlyContextVariables',
  'test.onlyStateVariables',
  'test.patternFiring',
  'test.patternFiringAad',
  'test.patternFiringAutoRecoverFalse',
  'test.patternLongRunning',
  'test.patternLongRunning.cancelAlertsOnRuleTimeout',
  'test.patternSuccessOrFailure',
  'test.restricted-noop',
  'test.throw',
  'test.unrestricted-noop',
  'test.validation'
```

```suggestion
  }
  return false;
};

const getRuleTypeIdLegacyConsumers = (ruleTypeId: string): string[] => {
  if (ruleTypeIdWithLegacyConsumers[ruleTypeId]) {
    return ruleTypeIdWithLegacyConsumers[ruleTypeId];
  }
```